### PR TITLE
Add support for Standalone Geyser (without Floodgate)

### PR DESCRIPTION
**Pull request by Smart123s**
### Summary of your change
[//]: # (Example: motiviation, enhancement)

I wanted to add support for servers with Geyser, but without Floodgate. Previously, they were all treated as Java players. Now they are not, and this gives us the ability to further configure Offline Geyser and Java name conflicts.

Notable changes in FastLogin's functionality:
- https://github.com/games647/FastLogin/commit/00196d8b1283f86b8646cb015763f2989ffc1c35, https://github.com/games647/FastLogin/commit/8d641a1fb6837311ca53a3dace35e5b349f9ae48: Created a superclass called BedrockService, so FloodgateService and GeyserService can share some code.
- https://github.com/games647/FastLogin/commit/3549465611c420785d4e134fe57469d2356f950b#diff-69f42c2b3e5137fc6ca05432413cc958fbf5fc288bf7ec872ce83e5374cfc507: Added the ability to resume Java checks for Geyser players
- https://github.com/games647/FastLogin/commit/06ef96f7cf656184bd59e550241a34266052306e: Added the ability to allow Geyser players with conflicting Java names to join the server, even if it would be prohibited by other configuration options in FastLogin (such as nameChangeCheck)

Minor changes (could have been in a separate PR, but I'm lazy):
- https://github.com/games647/FastLogin/commit/9c0c4cbfb44e37099510a3c6441a0254d004c6a5 and https://github.com/games647/FastLogin/commit/dd2dbdbc5452f62eb59d94542ac4a705cbcc8559

_For more details, read the individual commit messages._

### Related issue
Comments by @jondycz at
- https://github.com/games647/FastLogin/pull/635#issuecomment-950170109 -- and comments after this
- https://github.com/games647/FastLogin/issues/616#issuecomment-945169062  -- and comments after this

### Additional stuff
In config.yml, allowFloodgateNameConflict should now be called allowBedrockNameConflict, however changing that would break existing configurations. It would be ideal, to create a config migrator for this. Or any other ideas?

### CODE REVIEW

File: core/src/main/java/com/github/games647/fastlogin/core/hooks/bedrock/BedrockService.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import java.util.Optional;
+import java.util.UUID;
+
+import com.github.games647.craftapi.model.Profile;
+import com.github.games647.craftapi.resolver.RateLimitException;
+import com.github.games647.fastlogin.core.StoredProfile;
+import com.github.games647.fastlogin.core.shared.FastLoginCore;
+import com.github.games647.fastlogin.core.shared.LoginSource;
+
+//B is an instance of either FloodgatePlayer or GeyserSession
```
Comment (games647) [inline core/src/main/java/com/github/games647/fastlogin/core/hooks/bedrock/BedrockService.java:38]: Should be a JavaDoc comment.

**Review by games647 — CHANGES_REQUESTED** [review]

**Comment by jondycz** [comment]
> ### Summary of your change
> I wanted to add support for servers with Geyser, but without Floodgate. Previously, they were all treated as Java players. Now they are not, and this gives us the ability to further configure Offline Geyser and Java name conflicts.
> 
> Notable changes in FastLogin's functionality:
> 
> * [00196d8](https://github.com/games647/FastLogin/commit/00196d8b1283f86b8646cb015763f2989ffc1c35), [8d641a1](https://github.com/games647/FastLogin/commit/8d641a1fb6837311ca53a3dace35e5b349f9ae48): Created a superclass called BedrockService, so FloodgateService and GeyserService can share some code.
> * [3549465#diff-69f42c2b3e5137fc6ca05432413cc958fbf5fc288bf7ec872ce83e5374cfc507](https://github.com/games647/FastLogin/commit/3549465611c420785d4e134fe57469d2356f950b#diff-69f42c2b3e5137fc6ca05432413cc958fbf5fc288bf7ec872ce83e5374cfc507): Added the ability to resume Java checks for Geyser players
> * [06ef96f](https://github.com/games647/FastLogin/commit/06ef96f7cf656184bd59e550241a34266052306e): Added the ability to allow Geyser players with conflicting Java names to join the server, even if it would be prohibited by other configuration options in FastLogin (such as nameChangeCheck)
> 
> Minor changes (could have been in a separate PR, but I'm lazy):
> 
> * [9c0c4cb](https://github.com/games647/FastLogin/commit/9c0c4cbfb44e37099510a3c6441a0254d004c6a5) and [dd2dbdb](https://github.com/games647/FastLogin/commit/dd2dbdbc5452f62eb59d94542ac4a705cbcc8559)
> 
> _For more details, read the individual commit messages._
> 
> ### Related issue
> Comments by @jondycz at
> 
> * [Fix Floodgate player detection (caused by prefixes) #635 (comment)](https://github.com/games647/FastLogin/pull/635#issuecomment-950170109) -- and comments after this
> * [Error cannot reply to EncryptionRequestPacket on Floodgate players #616 (comment)](https://github.com/games647/FastLogin/issues/616#issuecomment-945169062)  -- and comments after this
> 
> ### Additional stuff
> In config.yml, allowFloodgateNameConflict should now be called allowBedrockNameConflict, however changing that would break existing configurations. It would be ideal, to create a config migrator for this. Or any other ideas?

Idea: Make allowFloodgateNameConflict an alias for allowBedrockNameConflict

File: core/src/main/java/com/github/games647/fastlogin/core/hooks/bedrock/BedrockService.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.UUID;
+
+import com.github.games647.craftapi.model.Profile;
+import com.github.games647.craftapi.resolver.RateLimitException;
+import com.github.games647.fastlogin.core.StoredProfile;
+import com.github.games647.fastlogin.core.shared.FastLoginCore;
+import com.github.games647.fastlogin.core.shared.LoginSource;
+
+/**
+ * B is an instance of either FloodgatePlayer or GeyserSession
```
Comment (games647) [inline core/src/main/java/com/github/games647/fastlogin/core/hooks/bedrock/BedrockService.java:39]:
```suggestion
 * @param <B> is an instance of either FloodgatePlayer or GeyserSession
```

**Review by games647 — CHANGES_REQUESTED** [review]

**Comment by Smart123s** [comment]
In the latest force-push, I have replaced this line:

https://github.com/games647/FastLogin/blob/a2ab79cd4112c314f28f16359c63d60c0c79ee1d/core/src/main/java/com/github/games647/fastlogin/core/hooks/bedrock/GeyserService.java#L69

with this:
```java
        for (GeyserSession gSess : geyser.getSessionManager().getSessions().values()) {
 ```

because the first one is no longer present in Geyser.

If something goes wrong, this might be the cause.

**Review by games647 — APPROVED** [review]

**Comment by games647** [comment]
Merged manually

**Pull request closed without merging**